opencog/cython/PythonEval.cc

```diff
@@ -1391,7 +1391,11 @@ void PythonEval::eval_expr_line(const std::string& partial_expr)
     // Ignore leading comments; don't ignore empty line.
     int c = 0;
     size_t part_size = part.size();
-    if (0 == part_size and 0 < partial_expr.size()) goto wait_for_more;
+    if (0 == part_size and 0 == partial_expr.size()) goto wait_for_more;
+    if (0 == part_size and
+        '\n' != partial_expr[0] and
+        '\r' != partial_expr[0])
+        goto wait_for_more;
```

Comment:
@linas I don't understand what you want to fix here, the former code `(0 < partion_exp.size())` looks perfectly good to me. The problem with your fix is that the following code
```python
def fun(x):
   y = x + 1

   return y
```
fails because `return y` is considered part of a subsequent block.